mender.go

```diff
@@ -525,23 +551,86 @@ func (m *mender) ReportUpdateStatus(update client.UpdateResponse, status string)
 	return nil
 }
 
-func reauthorize(m *mender) func() (client.AuthToken, error) {
+/* client closures */
+// see client.go: ApiRequest.Do()
+
+// reauthorize is a closure very similar to mender.Authorize(), but instead of
+// walking through all servers in the menderConfig.Servers list, it only tries
+// serverURL.
```

Comment:
I discovered a potential inconsistency if you were to simply call `mender.Authorize` here:
Since `Authorize` now inevitably starts from the beginning of the `Servers` list, if a request is rejected on the first server but is valid on the second server on the list but the `jwt` token happens to be expired, but authorization request is served by the first one. 
I changed this closure in order to only attempt to authorize on the current server in the `Do` loop.

Reply:
Ah, good catch!